commit 0fa32499118789b1ba92d7aed497449e2e2e3958
Author: q66 <q66@chimera-linux.org>
Date:   Sun Jul 23 22:46:01 2023 +0200

    handle utf-8 in qt streams in a qt6-compatible way

diff --git a/apps/launcher/maindialog.cpp b/apps/launcher/maindialog.cpp
index 69259f8..6990d69 100644
--- a/apps/launcher/maindialog.cpp
+++ b/apps/launcher/maindialog.cpp
@@ -2,13 +2,13 @@
 
 #include <components/version/version.hpp>
 #include <components/misc/helpviewer.hpp>
+#include <components/misc/utf8qtextstream.hpp>
 
 #include <QTime>
 #include <QDir>
 #include <QDebug>
 #include <QMessageBox>
 #include <QCloseEvent>
-#include <QTextCodec>
 
 #include <boost/program_options/options_description.hpp>
 #include <boost/program_options/variables_map.hpp>
@@ -316,7 +316,7 @@ bool Launcher::MainDialog::setupLauncherSettings()
                 return false;
             }
             QTextStream stream(&file);
-            stream.setCodec(QTextCodec::codecForName("UTF-8"));
+            ensureUtf8Encoding(stream);
 
             mLauncherSettings.readFile(stream);
         }
@@ -349,7 +349,7 @@ bool Launcher::MainDialog::setupGameSettings()
                 return {};
             }
             QTextStream stream(&file);
-            stream.setCodec(QTextCodec::codecForName("UTF-8"));
+            ensureUtf8Encoding(stream);
 
             (mGameSettings.*reader)(stream, ignoreContent);
             file.close();
@@ -536,7 +536,7 @@ bool Launcher::MainDialog::writeSettings()
 
     QTextStream stream(&file);
     stream.setDevice(&file);
-    stream.setCodec(QTextCodec::codecForName("UTF-8"));
+    ensureUtf8Encoding(stream);
 
     mLauncherSettings.writeFile(stream);
     file.close();
diff --git a/apps/wizard/inisettings.cpp b/apps/wizard/inisettings.cpp
index d3be2be..1e51d78 100644
--- a/apps/wizard/inisettings.cpp
+++ b/apps/wizard/inisettings.cpp
@@ -160,7 +160,7 @@ bool Wizard::IniSettings::writeFile(const QString &path, QTextStream &stream)
 
     if (file.open(QIODevice::ReadWrite | QIODevice::Truncate | QIODevice::Text)) {
         QTextStream in(&file);
-        in.setCodec(stream.codec());
+        in.setEncoding(stream.encoding());
 
         // Write the updated buffer to an empty file
         in << buffer;
diff --git a/apps/wizard/mainwizard.cpp b/apps/wizard/mainwizard.cpp
index 4648175..c8664ce 100644
--- a/apps/wizard/mainwizard.cpp
+++ b/apps/wizard/mainwizard.cpp
@@ -3,7 +3,6 @@
 #include <QDebug>
 #include <QDateTime>
 #include <QMessageBox>
-#include <QTextCodec>
 #include <QDir>
 
 #include "intropage.hpp"
@@ -19,6 +18,8 @@
 #include "installationpage.hpp"
 #endif
 
+#include <components/misc/utf8qtextstream.hpp>
+
 using namespace Process;
 
 Wizard::MainWizard::MainWizard(QWidget *parent) :
@@ -156,7 +157,7 @@ void Wizard::MainWizard::setupGameSettings()
             return;
         }
         QTextStream stream(&file);
-        stream.setCodec(QTextCodec::codecForName("UTF-8"));
+        ensureUtf8Encoding(stream);
 
         mGameSettings.readUserFile(stream);
     }
@@ -186,7 +187,7 @@ void Wizard::MainWizard::setupGameSettings()
                 return;
             }
             QTextStream stream(&file);
-            stream.setCodec(QTextCodec::codecForName("UTF-8"));
+            ensureUtf8Encoding(stream);
 
             mGameSettings.readFile(stream);
         }
@@ -220,7 +221,7 @@ void Wizard::MainWizard::setupLauncherSettings()
             return;
         }
         QTextStream stream(&file);
-        stream.setCodec(QTextCodec::codecForName("UTF-8"));
+        ensureUtf8Encoding(stream);
 
         mLauncherSettings.readFile(stream);
     }
@@ -428,7 +429,7 @@ void Wizard::MainWizard::writeSettings()
     }
 
     QTextStream stream(&file);
-    stream.setCodec(QTextCodec::codecForName("UTF-8"));
+    ensureUtf8Encoding(stream);
 
     mGameSettings.writeFile(stream);
     file.close();
@@ -451,7 +452,7 @@ void Wizard::MainWizard::writeSettings()
     }
 
     stream.setDevice(&file);
-    stream.setCodec(QTextCodec::codecForName("UTF-8"));
+    ensureUtf8Encoding(stream);
 
     mLauncherSettings.writeFile(stream);
     file.close();
diff --git a/components/CMakeLists.txt b/components/CMakeLists.txt
index 404ae4e..b909272 100644
--- a/components/CMakeLists.txt
+++ b/components/CMakeLists.txt
@@ -201,7 +201,7 @@ add_component_dir (esm4
 )
 
 add_component_dir (misc
-    constants utf8stream stringops resourcehelpers rng messageformatparser weakcache thread
+    constants utf8stream utf8qtextstream stringops resourcehelpers rng messageformatparser weakcache thread
     compression osguservalues errorMarker color
     )
 
diff --git a/components/config/gamesettings.cpp b/components/config/gamesettings.cpp
index 1c927d9..183a658 100644
--- a/components/config/gamesettings.cpp
+++ b/components/config/gamesettings.cpp
@@ -1,11 +1,11 @@
 #include "gamesettings.hpp"
 #include "launchersettings.hpp"
 
-#include <QTextCodec>
 #include <QDir>
 #include <QRegularExpression>
 
 #include <components/files/configurationmanager.hpp>
+#include <components/misc/utf8qtextstream.hpp>
 
 const char Config::GameSettings::sArchiveKey[] = "fallback-archive";
 const char Config::GameSettings::sContentKey[] = "content";
@@ -235,7 +235,7 @@ bool Config::GameSettings::isOrderedLine(const QString& line)
 bool Config::GameSettings::writeFileWithComments(QFile &file)
 {
     QTextStream stream(&file);
-    stream.setCodec(QTextCodec::codecForName("UTF-8"));
+    ensureUtf8Encoding(stream);
 
     // slurp
     std::vector<QString> fileCopy;
diff --git a/components/misc/utf8qtextstream.hpp b/components/misc/utf8qtextstream.hpp
new file mode 100644
index 0000000..37ee2cf
--- /dev/null
+++ b/components/misc/utf8qtextstream.hpp
@@ -0,0 +1,20 @@
+#ifndef MISC_UTF8QTEXTSTREAM_HPP
+#define MISC_UTF8QTEXTSTREAM_HPP
+
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+#include <QTextCodec>
+#endif
+#include <QTextStream>
+
+namespace
+{
+    void ensureUtf8Encoding(QTextStream& stream)
+    {
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+        stream.setCodec(QTextCodec::codecForName("UTF-8"));
+#else
+        stream.setEncoding(QStringConverter::Utf8);
+#endif
+    }
+}
+#endif
